src/main/java/com/faforever/client/vault/SpecialCaseForMapVaultLocationChecker.java

```diff
@@ -0,0 +1,60 @@
+package com.faforever.client.vault;
+
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.notification.Action;
+import com.faforever.client.notification.NotificationService;
+import com.faforever.client.notification.PersistentNotification;
+import com.faforever.client.notification.Severity;
+import com.faforever.client.preferences.ForgedAlliancePrefs;
+import com.faforever.client.preferences.PreferencesService;
+import com.google.common.base.CharMatcher;
+import com.sun.security.auth.module.NTSystem;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+
+import static org.bridj.Platform.isUnix;
+import static org.bridj.Platform.isWindows;
+
+/**
+ * Checks if the user is on linux or has a none ascii username. In this cases we need to use a fallback location for the
+ * vault.
+ */
+@Component
+public class SpecialCaseForMapVaultLocationChecker implements InitializingBean {
+  public static final String ONE_DRIVE_IDENTIFIER = "one";
```

Comment:
not sure how the onedrive folder is called

Reply:
still to be discussed

Reply:
I suggest matching against this regex: `[Oo]ne[ ]*[Dd]rive`

Reply:
example of one user who had the client store stuff in the onedrive folder `C:\Users\%Username%\OneDrive\Documents\My Games\Gas Powered Games\Supreme Commander Forged Alliance\mods`